backend/remote-state/cos/backend_test.go

```diff
@@ -0,0 +1,226 @@
+package cos
+
+import (
+	"crypto/md5"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/hashicorp/terraform/backend"
+	"github.com/hashicorp/terraform/state/remote"
+	"github.com/likexian/gokit/assert"
+)
+
+const (
+	appId         = 1258798060
```

Comment:
It looks like this `appId` is necessary to access/run these tests -- may I suggest making this configurable through an environment variable, such that anyone could run the tests by updating it without modifying the source code?

Reply:
You are right, I have move it to `TF_COS_APPID` environment variables